### DISCUSSION TITLE
How to query a mongodb view

### DISCUSSION THREAD

**Discussion by chadacious** [typegoose/typegoose#738] [Q&A] [answered]
I haven't found any documentation, stackoverflow or github issue discussion on how one might query a view in mongodb with typegoose.

Is it possible? If so, could an example be provided?

**Comment by hasezoey** [comment]
Typegoose does not change anything regarding querys, that is still native mongoose - so you would be better off searching with term "mongoose" instead of "typegoose"

---

Also from what i can tell in mongoose a view query is done like a normal mongoose schema & model & query

The best example i could find is [in this SO question / answer](https://stackoverflow.com/questions/68705163/querying-views-with-mongoose-and-encapsulating-documents-fields)

**Comment by chadacious** [comment, marked as answer]
In my case, I had a view that did a complex series of lookups and matches to ultimately project the same schema as another collection. By specifying schemaOptions for a new model, I could specify the view name I had created in my mongodb:

```
import { getModelForClass, modelOptions } from "@typegoose/typegoose";

@modelOptions({ schemaOptions: { collection: "view_name" } })
export class NewClassName extends TheBaseClass { }

export const NewClassNameModel = getModelForClass(NewClassName);
```
